pallets/loans/src/lib.rs

```diff
@@ -1011,6 +1209,188 @@ impl<T: Config> Pallet<T> {
+
+                RewardAccured::<T>::try_mutate(supplier, |total_reward| -> DispatchResult {
+                    let supplier_account = AccountDeposits::<T>::get(asset_id, supplier);
+                    let supplier_amount = supplier_account.voucher_balance;
+                    let reward_delta = Self::calculate_reward_delta(supplier_amount, delta_index)?;
+                    *total_reward = total_reward
+                        .checked_add(reward_delta)
+                        .ok_or(ArithmeticError::Overflow)?;
+                    Self::deposit_event(Event::<T>::DistributedSupplierReward(
+                        asset_id,
+                        supplier.clone(),
+                        reward_delta,
+                        supply_state.index,
+                    ));
+
+                    Ok(())
+                })
+            },
+        )
+    }
+
+    fn distribute_borrower_reward(
+        asset_id: AssetIdOf<T>,
+        borrower: &T::AccountId,
+    ) -> DispatchResult {
+        RewardBorrowerIndex::<T>::try_mutate(
+            asset_id,
+            borrower,
+            |borrower_index| -> DispatchResult {
+                let borrow_state = RewardBorrowState::<T>::get(asset_id);
+                let delta_index = borrow_state
+                    .index
+                    .checked_sub(*borrower_index)
+                    .ok_or(ArithmeticError::Underflow)?;
+                *borrower_index = borrow_state.index;
+
+                RewardAccured::<T>::try_mutate(borrower, |total_reward| -> DispatchResult {
+                    let current_borrow_amount = Self::current_borrow_balance(borrower, asset_id)?;
+                    let reward_delta =
+                        Self::calculate_reward_delta(current_borrow_amount, delta_index)?;
+                    *total_reward = total_reward
+                        .checked_add(reward_delta)
+                        .ok_or(ArithmeticError::Overflow)?;
+                    Self::deposit_event(Event::<T>::DistributedSupplierReward(
+                        asset_id,
+                        borrower.clone(),
+                        reward_delta,
+                        borrow_state.index,
+                    ));
+
+                    Ok(())
+                })
+            },
+        )
+    }
+
+    fn collect_market_reward(asset_id: AssetIdOf<T>, user: &T::AccountId) -> DispatchResult {
+        Self::update_reward_supply_index(asset_id)?;
+        Self::distribute_supplier_reward(asset_id, user)?;
+
+        Self::update_reward_borrow_index(asset_id)?;
+        Self::distribute_borrower_reward(asset_id, user)?;
+
+        Ok(())
+    }
+
+    fn pay_reward(user: &T::AccountId) -> DispatchResult {
+        let pool_account = Self::account_id();
+        let reward_asset = T::RewardAssetId::get();
+        RewardAccured::<T>::try_mutate(user, |total_reward| -> DispatchResult {
+            let total = *total_reward;
+            if total > 0 {
```

Comment:
not necessary to have this `total` alias

Reply:
total is used by deposit_event.